File: node/verify_setup.js
#!/usr/bin/env node
/**
 * Run this to verify your environment is correctly configured.
 */

import 'dotenv/config';

const baseUrl = process.env.OLLAMA_BASE_URL || 'http://localhost:11434';
const model = process.env.MODEL_NAME || 'qwen3.5:4b';

console.log(`Checking Ollama at ${baseUrl}...`);

try {
  // Check Ollama is running
  const tagsResp = await fetch(`${baseUrl}/api/tags`);
  const tagsData = await tagsResp.json();
  const modelNames = (tagsData.models || []).map((m) => m.name);
  console.log(`  Available models: ${JSON.stringify(modelNames)}`);

  // Check our model is available
  if (!modelNames.some((name) => name.includes(model))) {
    console.log(`  ERROR: Model '${model}' not found. Run: ollama pull ${model}`);
    process.exit(1);
  }

  // Test inference
  console.log(`Testing inference with ${model}...`);
  const chatResp = await fetch(`${baseUrl}/api/chat`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({
      model,
      messages: [{ role: 'user', content: 'Respond with: VERIFIED' }],
      stream: false,
      options: { temperature: 0.0 },
    }),
  });
  const chatData = await chatResp.json();
  console.log(`  Model response: ${chatData.message.content}`);

  console.log('\nAll checks passed. You\'re ready for the lab!');
} catch (e) {
  console.log(`  ERROR: Cannot reach Ollama — ${e.message}`);
  console.log('  Make sure \'ollama serve\' is running.');
  process.exit(1);
}

File: python/verify_setup.py
#!/usr/bin/env python3
"""Run this to verify your environment is correctly configured."""

import asyncio
import os

import httpx
from dotenv import load_dotenv

load_dotenv()


async def verify():
    base_url = os.getenv("OLLAMA_BASE_URL", "http://localhost:11434")
    model = os.getenv("MODEL_NAME", "qwen3.5:4b")

    print(f"Checking Ollama at {base_url}...")

    async with httpx.AsyncClient(timeout=30.0) as client:
        # Check Ollama is running
        try:
            resp = await client.get(f"{base_url}/api/tags")
            models = resp.json().get("models", [])
            model_names = [m["name"] for m in models]
            print(f"  Available models: {model_names}")
        except Exception as e:
            print(f"  ERROR: Cannot reach Ollama — {e}")
            print("  Make sure 'ollama serve' is running.")
            return False

        # Check our model is available
        if not any(model in name for name in model_names):
            print(f"  ERROR: Model '{model}' not found. Run: ollama pull {model}")
            return False

        # Test inference
        print(f"Testing inference with {model}...")
        resp = await client.post(
            f"{base_url}/api/chat",
            json={
                "model": model,
                "messages": [{"role": "user", "content": "Respond with: VERIFIED"}],
                "stream": False,
                "options": {"temperature": 0.0},
            },
        )
        result = resp.json()
        content = result["message"]["content"]
        print(f"  Model response: {content}")

    print("\nAll checks passed. You're ready for the lab!")
    return True


if __name__ == "__main__":
    asyncio.run(verify())
